pkg/db/end_users.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package db
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+const (
+	sqlCreateEndUser = `
+CREATE TABLE end_user (
+    id int PRIMARY KEY AUTO_INCREMENT,
+    sub VARCHAR(2000) NOT NULL
+)`
+	sqlInsertEndUser = `insert into end_user (sub) values (?)`
+)
+
+// EndUser is human user operating the User Agent and the Web Wallet.
+type EndUser struct {
+	ID  int64
+	Sub string
+}
+
+// EndUsers is an EndUser DAO.
+type EndUsers struct {
```

Comment:
Shouldn't we have a configurable storage and sql could be one of the implementation ? This is tightly coupled to mysql now.

Reply:
yes

Reply:
It's coupled to an RDMS, yes. What would a generic interface that encompasses SQL and NoSQL look like?